flink-table/flink-sql-parser/src/test/java/org/apache/flink/sql/parser/FlinkDDLDataTypeTest.java

```diff
@@ -0,0 +1,441 @@
+import org.apache.calcite.rel.type.RelDataTypeFactory;
+import org.apache.calcite.rel.type.RelDataTypeSystem;
+import org.apache.calcite.sql.SqlDataTypeSpec;
+import org.apache.calcite.sql.SqlDialect;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlNodeList;
+import org.apache.calcite.sql.SqlWriter;
+import org.apache.calcite.sql.dialect.CalciteSqlDialect;
+import org.apache.calcite.sql.parser.SqlParseException;
+import org.apache.calcite.sql.parser.SqlParser;
+import org.apache.calcite.sql.parser.SqlParserTest;
+import org.apache.calcite.sql.parser.SqlParserUtil;
+import org.apache.calcite.sql.pretty.SqlPrettyWriter;
+import org.apache.calcite.sql.type.SqlTypeFactoryImpl;
+import org.apache.calcite.sql.type.SqlTypeName;
+import org.apache.calcite.test.SqlValidatorTestCase;
+import org.apache.calcite.util.Util;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import javax.annotation.Nullable;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Tests for all the sup[ported flink DDL data types.
+ */
+@RunWith(Parameterized.class)
+public class FlinkDDLDataTypeTest {
+	private FlinkSqlConformance conformance = FlinkSqlConformance.DEFAULT;
+	private static final RelDataTypeFactory TYPE_FACTORY =
+		new SqlTypeFactoryImpl(RelDataTypeSystem.DEFAULT);
+	private static final Fixture FIXTURE = new Fixture(TYPE_FACTORY);
+	private static final String DDL_FORMAT = "create table t1 (\n" +
+		"  f0 %s\n" +
+		") with (\n" +
+		"  k1 = 'v1'\n" +
+		")";
+
+	@Parameterized.Parameters(name = "{index}: {0}")
+	public static List<TestItem> testData() {
+		return Arrays.asList(
+			createTestItem("CHAR", nullable(FIXTURE.char1Type), "CHAR"),
+			createTestItem("CHAR NOT NULL", FIXTURE.char1Type, "CHAR NOT NULL"),
+			createTestItem("CHAR   NOT \t\nNULL", FIXTURE.char1Type, "CHAR NOT NULL"),
+			createTestItem("char not null", FIXTURE.char1Type, "CHAR NOT NULL"),
+			createTestItem("CHAR NULL", nullable(FIXTURE.char1Type), "CHAR"),
+			createTestItem("CHAR(33)", nullable(FIXTURE.char33Type), "CHAR(33)"),
+			createTestItem("VARCHAR", nullable(FIXTURE.varcharType), "VARCHAR"),
+			createTestItem("VARCHAR(33)", nullable(FIXTURE.varchar33Type), "VARCHAR(33)"),
+			createTestItem("STRING",
+				nullable(FIXTURE.createSqlType(SqlTypeName.VARCHAR, Integer.MAX_VALUE)), "STRING"),
+			createTestItem("BOOLEAN", nullable(FIXTURE.booleanType), "BOOLEAN"),
+			createTestItem("BINARY", nullable(FIXTURE.binaryType), "BINARY"),
+			createTestItem("BINARY(33)", nullable(FIXTURE.binary33Type), "BINARY(33)"),
+			createTestItem("VARBINARY", nullable(FIXTURE.varbinaryType), "VARBINARY"),
+			createTestItem("VARBINARY(33)", nullable(FIXTURE.varbinary33Type),
+				"VARBINARY(33)"),
+			createTestItem("BYTES",
+				nullable(FIXTURE.createSqlType(SqlTypeName.VARBINARY, Integer.MAX_VALUE)),
```

Comment:
Add this to `FIXTURE` as well?

Reply:
I put it here intentionally, so we can see that this type has the max precision.